### PR TITLE
VQA task guide

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR adds a new Visual Question Answering task guide to the transformers docs: 
fine-tuning ViLT, based on @NielsRogge 's [notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/ViLT/Fine_tuning_ViLT_for_VQA.ipynb)

### CODE REVIEW

**Comment by MKhalusova** [comment]
Image is in [this PR](https://huggingface.co/datasets/huggingface/documentation-images/discussions/149)

File: docs/source/en/tasks/visual_question_answering.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+... )
+```
+
+3.Call `train` to finetune your model.
+
+```py
+>>> trainer.train() 
+```
+
+Once training is completed, share your model to the Hub with the `push_to_hub` method to share your final model on the 🤗 Hub:
```
Comment (stevhliu) [inline docs/source/en/tasks/visual_question_answering.md:284]:
```suggestion
Once training is completed, share your model to the Hub with the [`~Trainer.push_to_hub`] method to share your final model on the 🤗 Hub:
```

**Review by stevhliu — APPROVED** [review]
Really nice! ❤️

File: docs/source/en/tasks/visual_question_answering.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+In this guide you'll learn how to:
+
+- Fine-tune a VQA model, specifically [ViLT](../model_doc/vilt), on the [`Graphcore/vqa` dataset](https://huggingface.co/datasets/Graphcore/vqa).
+- Use your fine-tuned model for inference.
+
+ViLT model incorporates text embeddings into a Vision Transformer (ViT), allowing it to have a minimal design for 
+Vision-and-Language Pre-training (VLP). This model can be used for several downstream tasks. For the VQA task, a classifier 
+head is placed on top (a linear layer on top of the final hidden state of the `[CLS]` token) and randomly initialized. 
+Visual Question Answering is thus treated as a classification problem.
```
Comment (NielsRogge) [inline docs/source/en/tasks/visual_question_answering.md:39]: As discussed offline, would definitely be great to emphasize here that ViLT is one of the few models in the library that treats VQA as a classification tasks, all the newer models (e.g. BLIP, InstructBLIP) treat it as a generative task.

File: docs/source/en/tasks/visual_question_answering.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+Let's define the model checkpoint as a global variable.
+
+```py
+>>> model_checkpoint = "dandelin/vilt-b32-mlm"
+```
+
+## Load the data
+
+For illustration purposes, in this guide we use a very small sample of the annotated visual question answering `Graphcore/vqa` dataset.
```
Comment (NielsRogge) [inline docs/source/en/tasks/visual_question_answering.md:64]: As always, many people say to me "tutorials show how to do it on a dataset from the hub but what if I want to apply this on my own data?"

Hence would be great to link to a guide explaining how to load custom data into a 🤗 Dataset

File: docs/source/en/tasks/visual_question_answering.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+>>> processor = ViltProcessor.from_pretrained(model_checkpoint)
+```
+
+To preprocess the data we need to encode the images and questions using the [`ViltProcessor`]. The processor will use 
+the [`BertTokenizerFast`] to tokenize the text and create `input_ids`, `attention_mask` and `token_type_ids` for the text data. 
+As for images, the processor will leverage [`ViltImageProcessor`] to resize and normalize the image, and create `pixel_values` and `pixel_mask`.
+
+All these preprocessing steps are done under the hood, we only need to call the `processor`. However, we still need to 
+prepare the target labels. Labels are represented as a soft encoded vector where of shape (num_labels,) where the each
```
Comment (NielsRogge) [inline docs/source/en/tasks/visual_question_answering.md:178]:
```suggestion
All these preprocessing steps are done under the hood, we only need to call the `processor`. However, we still need to 
prepare the target labels. Labels are represented as a soft encoded vector of shape (num_labels,) where the each 
```

File: docs/source/en/tasks/visual_question_answering.md [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+>>> processor = ViltProcessor.from_pretrained(model_checkpoint)
+```
+
+To preprocess the data we need to encode the images and questions using the [`ViltProcessor`]. The processor will use 
+the [`BertTokenizerFast`] to tokenize the text and create `input_ids`, `attention_mask` and `token_type_ids` for the text data. 
+As for images, the processor will leverage [`ViltImageProcessor`] to resize and normalize the image, and create `pixel_values` and `pixel_mask`.
+
+All these preprocessing steps are done under the hood, we only need to call the `processor`. However, we still need to 
+prepare the target labels. Labels are represented as a soft encoded vector where of shape (num_labels,) where the each 
+element represents a potential answer (label): for valid answers the element contains their score (weight), and the rest are zeroes.
```
Comment (NielsRogge) [inline docs/source/en/tasks/visual_question_answering.md:179]: The last part here doesn't seem part of the sentence

**Review by NielsRogge — APPROVED** [review]
Thanks for working on this! Would be great to have a guide on VQA with newer generative models as well.

File: docs/source/en/tasks/visual_question_answering.md (hunk 3 above)
Comment (MKhalusova) [reply docs/source/en/tasks/visual_question_answering.md:70]: Linked to a guide on how to load custom image data into a Dataset

File: docs/source/en/tasks/visual_question_answering.md (hunk 2 above)
Comment (MKhalusova) [reply docs/source/en/tasks/visual_question_answering.md]: Added more clarifications, and an example on how to do VQA inference with BLIP-2

File: docs/source/en/tasks/visual_question_answering.md [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+
+Some noteworthy use case examples for VQA include:
+* Accessibility applications for visually impaired individuals.
+* Education: posing questions about visual materials presented in lectures or textbooks. VQA can also be utilized in interactive museum exhibits or historical sites.
+* Customer service and e-commerce: VQA can enhance user experience by letting users ask questions about products. 
+* Image retrieval: VQA models can be used to retrieve images with specific characteristics. For example, the user can ask "Is there a dog?" to find all images with dogs from a set of images.
+
+In this guide you'll learn how to:
+
+- Fine-tune a VQA model, specifically [ViLT](../model_doc/vilt), on the [`Graphcore/vqa` dataset](https://huggingface.co/datasets/Graphcore/vqa).
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/tasks/visual_question_answering.md]: Addressed - added a comment where to download the data, linked to a guide how to create an image dataset. 
Plus, I created my own dataset with the same script, so should something happen (we’ve got tests), I can make a quick substitute. Until then, I don’t see an issue - we already use third-party datasets in other task guides (e.g."lambdalabs/pokemon-blip-captions”, "PolyAI/minds14”, "facebook/voxpopuli”).

**Comment by MKhalusova** [comment]
Addressed @NielsRogge's and @Steven's feedback. @amyeroberts please take a look :)

File: docs/source/en/tasks/visual_question_answering.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+same data manually from the official [VQA dataset page](https://visualqa.org/download.html). If you prefer to follow the 
+tutorial with your custom data, check out how to [Create an image dataset](https://huggingface.co/docs/datasets/image_dataset#loading-script)
+guide in the 🤗 Datasets documentation.  
+
+Let's load the first 200 examples from the validation split and explore the dataset's features:  
+
+```python
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("Graphcore/vqa", split="validation[:200]")
```
Comment (amyeroberts) [inline docs/source/en/tasks/visual_question_answering.md:83]: Why are we training on the validation split?

File: docs/source/en/tasks/visual_question_answering.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+```py
+>>> from transformers import pipeline
+
+>>> pipe = pipeline("visual-question-answering", model="MariaK/vilt_finetuned_200")
+```
+
+The model in this guide has only been trained on 200 examples, so don't expect a lot of it. Let's see if it at least 
+learned something from the data and take the first example from the dataset to illustrate inference:
+
+```py
```
Comment (amyeroberts) [inline docs/source/en/tasks/visual_question_answering.md:315]: nice! 🤗

File: docs/source/en/tasks/visual_question_answering.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+Now that you have fine-tuned a ViLT model, and uploaded it to the 🤗 Hub, you can use it for inference. The simplest
+way to try out your fine-tuned model for inference is to use it in a [`Pipeline`].
+
+```py
+>>> from transformers import pipeline
+
+>>> pipe = pipeline("visual-question-answering", model="MariaK/vilt_finetuned_200")
+```
+
+The model in this guide has only been trained on 200 examples, so don't expect a lot of it. Let's see if it at least
```
Comment (amyeroberts) [inline docs/source/en/tasks/visual_question_answering.md:312]: nit: with 'of' it parses to me that not much of the model will be returned

```suggestion
The model in this guide has only been trained on 200 examples, so don't expect a lot from it. Let's see if it at least 
```

File: docs/source/en/tasks/visual_question_answering.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+Let's illustrate how you can use this model for VQA. First, let's load the model and leverage a GPU: 
+
+```py
+>>> from transformers import AutoProcessor, Blip2ForConditionalGeneration
+>>> import torch
+
+>>> processor = AutoProcessor.from_pretrained("Salesforce/blip2-opt-2.7b")
+>>> model = Blip2ForConditionalGeneration.from_pretrained("Salesforce/blip2-opt-2.7b", torch_dtype=torch.float16)
+>>> device = "cuda" if torch.cuda.is_available() else "cpu"
+>>> model.to(device)
```
Comment (amyeroberts) [inline docs/source/en/tasks/visual_question_answering.md:369]: Maybe we should clarify here that we set explicitly to GPU if available here, but it's automatically handled by the trainer?

**Review by amyeroberts — APPROVED** [review]
Nice! Thanks for adding this guide :) 

Just a few nits on my side.
